magma/lib/magma/query/predicate/with_date_time_predicate_methods.rb

```diff
@@ -0,0 +1,7 @@
+class Magma
```

Comment:
Not sure I understand how these modules are used - why can't these just be in the DateTimePredicate itself?

Reply:
They could be left there, but then they would have to be duplicated in the `date_time_subselect.rb` class, too. Otherwise I think the verb actions that call those methods would not work?

Reply:
Generally, the new subselect predicates copy the verbs from the regular predicates, but are subclasses of a new `ColumnSubselectPredicate`, and they share different `#extract` code to handle the subselect nested data. So I was treating these modules as mixins, where you need the same methods and verbs for regular predicates and subselect predicates. 

I supposed I could also organize it the other way, subclass the subselect predicates like `DateTimeSubselectPredicate < DateTimePredicate`, and include a module for `WithSubselectExtract` that just overrides `#extract`. Would that be better?

Reply:
I have the vague impression that the SubselectPredicate classes could all be collapsed into their counterpart, and instead have the various `select` and `extract` bits for on the basis of something like `#is_subselect?`, which might be a little more compact. Could that work?

Reply:
That's the way I originally worked it out when exploring this idea, but I think that is less clear and maintainable, since you wind up having a ton of `is_subselect` `if` statements laying around, and you have to pass that flag around quite a bit. It's less extensible (if we wind up having a third "subtype" of predicate, for example), and would seem to violate the single-responsibility principle (the same class / method would be doing two different things).

By organizing all the subselect-related code into their own set of classes, I think that will help isolate any bugs and changes that would be related to the feature.